aci/resource_aci_dhcpoptionpol.go

```diff
@@ -296,50 +292,47 @@ func resourceAciDHCPOptionPolicyUpdate(ctx context.Context, d *schema.ResourceDa
 	}
 
 	if d.HasChange("dhcp_option") {
-		dhcpOption := d.Get("dhcp_option_ids").([]interface{})
-		for _, val := range dhcpOption {
-			dhcpOptionDN := val.(string)
-			err := aciClient.DeleteByDn(dhcpOptionDN, "dhcpOption")
+		old_options, _ := d.GetChange("dhcp_option")
+
+		for _, optionParam := range old_options.([]interface{}) {
+			optionMap := optionParam.(map[string]interface{})
+			err := aciClient.DeleteByDn(optionMap["id"].(string), "dhcpOption")
```

Comment:
Can we check if they are actually different before deleting them?

Reply:
if d.HasChange("dhcp_option") ensures that the option is different?